File: server/src/main/java/ch/ethz/lapis/api/sql/Query.java
package ch.ethz.lapis.api.sql;

import lombok.Data;
import net.sf.jsqlparser.expression.Expression;

import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@Data
public class Query implements Cloneable {

    private String table;
    private Map<String, String> aliasToExpression = new HashMap<>();
    private List<String> selectExpressions = new ArrayList<>();
    private Expression whereExpression;
    private List<String> groupByColumns = new ArrayList<>();
    private Expression havingExpression;
    private String orderByExpression;
    private boolean orderByAsc = true;
    private Integer offset;
    private Integer limit;

    @Override
    protected Query clone() {
        try {
            return (Query) super.clone();
        } catch (CloneNotSupportedException e) {
            throw new RuntimeException(e);
        }
    }
}

File: lapis2/src/main/kotlin/org/genspectrum/lapis/sqlForChat/Query.kt
package org.genspectrum.lapis.sqlForChat

import net.sf.jsqlparser.expression.Expression
import org.genspectrum.lapis.silo.SiloFilterExpression

data class Query (
    var table: String? = null,
    var aliasToExpression: Map<String, String> = mutableMapOf(),
    var selectExpressions: MutableList<String> = mutableListOf(),
    var whereExpression: Expression? = null, // The jsqlparser version of the expression
    var whereQueryExpr: SiloFilterExpression? = null, // Our version of the expression
    var groupByColumns: MutableList<String> = mutableListOf(),
    var havingExpression: Expression? = null,
    var orderByExpression: String? = null,
    var orderByAsc: Boolean = true,
    var offset: Int? = null,
    var limit: Int? = null,
)
